Carbon.Security/Functions/ConvertTo-AesKey.ps1

```diff
@@ -0,0 +1,59 @@
+
+function ConvertTo-AesKey
+{
+    [CmdletBinding()]
+    [OutputType([byte[]])]
+    param(
+        [Parameter(Mandatory)]
+        [String]$From,
+
+        [Parameter(Mandatory)]
+        [Object]$InputObject
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState    
+
+    $Key = $InputObject
+
+    if( $InputObject -isnot [byte[]] )
+    {
+        if( $InputObject -is [SecureString] )
+        {
+            $unicodeKey = Convert-CSecureStringToByte -SecureString $InputObject
+            try
+            {
+                # SecureString is two bytes per char. We need an encoding that is one byte per char, otherwise the key
+                # will be twice as big as it should be.
+                $Key = [Text.Encoding]::Convert([Text.Encoding]::Unicode, [Text.Encoding]::UTF8, $unicodeKey)

```

Comment:
So, I left converting it to UTF8 because most of the time, that's going to result in a 1-byte per character string. I didn't want to restrict users to just the ASCII character set, so I expanded the documentation to explain what the secure string should contain, with code to help people figure out how long it is (in bytes). I also improved the error message when a converted secure string key ends up being the wrong length.

Reply:
👍